src/QMCDrivers/DMC/DMCOMP.cpp

```diff
@@ -298,9 +283,16 @@ bool DMCOMP::run()
         Movers[ip]->setMultiplicity(wit,wit_end);
         if(QMCDriverMode[QMC_UPDATE_MODE] && now%updatePeriod == 0)
           Movers[ip]->updateWalkers(wit, wit_end);
-        // recompute the accuracy critical part of Psi at the end of the last step.
-        if ( step+1 == nSteps && nBlocksBetweenRecompute && (1+block)%nBlocksBetweenRecompute == 0 && QMCDriverMode[QMC_UPDATE_MODE] )
-          Movers[ip]->recomputePsi(wit,wit_end);
+#endif
+        wClones[ip]->resetCollectables();
+        const size_t nw=W.getActiveWalkers();
+#pragma omp for nowait
+        for(size_t iw=0;iw<nw; ++iw)
+        {
+          Walker_t& thisWalker(*W.getWalker(iw));
+          Movers[ip]->advanceWalker(thisWalker,recompute);
```

Comment:
This is to keep DMC/VMCUpdatePbyP/All. The manipulation of walkers is all handled in the direct VMC/DMC driver.